provider/datagen/tests/testdata/work_log+keys.txt

```diff
@@ -1,5 +0,0 @@
-datetime/lengths@1
```

Comment:
Issue: Please restore this file and ensure that `work_log.wasm` is unchanged across the PR.

Reply:
What for? I've removed the key-file functionality from datagen as users can provide a list of keys as either a `&[ResourceKey]` or `&[&str]`, and it's not needed as an intermediate format for key extraction anymore either.

Reply:
The key file tests two things:

1. That key extraction works
2. That compiling the binary with the appropriate optimization options produces the same key list, and that we are aware when the key list changes (i.e., a golden data test)

(1) is tested elsewhere now, which is fine, but not (2) I think?

Reply:
OK, it looks like (2) is tested.

Why did the WASM file have a +2MB change, then?

Reply:
Because it's new. Before we had checked in the key file, now we're checking in the binary